perfkitbenchmarker/providers/docker/docker_virtual_machine.py

```diff
@@ -0,0 +1,303 @@
+    self.container_image = DEFAULT_DOCKER_IMAGE
+    self.docker_sysctl_flags = ''
+    # apply flags
+    if FLAGS.docker_custom_image:
+      self.container_image = FLAGS.docker_custom_image
+    if FLAGS.docker_sysctl_flags:
+      self.docker_sysctl_flags = FLAGS.docker_sysctl_flags
+
+  def _CreateDependencies(self):
+    self._CreateVolumes()
+
+  def _DeleteDependencies(self):
+    self._DeleteVolumes()
+
+  def _Create(self):
+    """Create a Docker instance"""
+
+    # Locally build docker container
+    with self.docker_build_lock:
+      image_exists = self._LocalImageExists(self.container_image)
+      if image_exists is False:
+        self._BuildImageLocally()
+
+    create_command = self._FormatCreateCommand()
+    container_info, _, _ = vm_util.IssueCommand(create_command)
+    self.container_id = container_info.encode("ascii")
+
+  def _FormatCreateCommand(self):
+    """Formats the command for Docker based on vm_spec and flags"""
+
+    create_command = ['docker', 'run', '-d', '--name', self.name]
+
+    # format scratch disks
+    for vol in self.scratch_disks:
+      vol_string = vol.volume_name + ":" + vol.mount_point
+      create_command.append('-v')
+      create_command.append(vol_string)
+    # format cpus option
+    if self.cpus > 0:
+      create_command.append('--cpus')
+      create_command.append(self.cpus)
+    # format memory option
+    if self.memory_mb > 0:
+      create_command.append('-m')
+      create_command.append(str(self.memory_mb) + 'm')
+    if self.docker_sysctl_flags:
+      create_command.append('--sysctl')
+      create_command.append(self.docker_sysctl_flags)
+
+    create_command.append(self.container_image)
+    create_command.append('/usr/sbin/sshd')
+    create_command.append('-D')
+
+    return create_command
+
+  @vm_util.Retry()
+  def _PostCreate(self):
+    """
+    Prepares running container. Gets the IP address, copies public keys,
+    and configures the proxy if one is specified
+    """
+    self._GetIpAddresses()
+
+    # Copy ssh key to container to enable ssh login
+    copy_ssh_command = ['docker', 'cp', self.ssh_public_key,
+                        '%s:/root/.ssh/authorized_keys' % self.name]
+    vm_util.IssueCommand(copy_ssh_command)
+
+    # change ownership of authorized_key file to root in container
+    chown_command = ['docker', 'exec', self.name, 'chown',
+                     'root:root', '/root/.ssh/authorized_keys']
+    vm_util.IssueCommand(chown_command)
+    self._ConfigureProxy()
+
+  def _Delete(self):
+    """Kill and Remove Docker Container"""
+
+    delete_command = ['docker', 'kill', self.name]
+    output = vm_util.IssueCommand(delete_command)
+    logging.info(output[STDOUT].rstrip())
+
+    remove_command = ['docker', 'rm', self.name]
+    output = vm_util.IssueCommand(remove_command)
+    logging.info(output[STDOUT].rstrip())
+
+    return
+
+  @vm_util.Retry(poll_interval=10, max_retries=10)
+  def _Exists(self):
+    """
```

Comment:
There is a bit of inconsistency in the docstring style. See https://www.python.org/dev/peps/pep-0257/ for formatting of single line and multiline docstrings and when to use either. In particular, single line docstring comments should be on the same line as the """ delimiters.

A single line here might be:
"""Returns whether the container is up and running."""